### PR TITLE
feat: support health check on ApisixUpstream

### PR DESCRIPTION
**Pull request by tokers**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

**Comment by tokers** [comment]
Test cases will be added as soon as possible.

File: pkg/apisix/resource.go [hunk 1]
```diff
@@ -135,6 +135,17 @@ func (i *item) upstream(clusterName string) (*v1.Upstream, error) {
 		})
 	}
 
+	// XXX This is a work around scheme to avoid APISIX's
```
Comment (gxthrj) [inline pkg/apisix/resource.go:138]: typo `XXX `

File: pkg/apisix/resource.go [hunk 2]
```diff
@@ -135,6 +135,17 @@ func (i *item) upstream(clusterName string) (*v1.Upstream, error) {
...
+	// health check schema about the health checker intervals.
+	if ups.Checks != nil && ups.Checks.Active != nil {
+		if ups.Checks.Active.Healthy.Interval == 0 {
+			ups.Checks.Active.Healthy.Interval = 1
```
Comment (gxthrj) [inline pkg/apisix/resource.go:142]: Magic number, need a default interval variable instead.

File: pkg/apisix/resource.go [hunk 3]
```diff
@@ -135,6 +135,17 @@ func (i *item) upstream(clusterName string) (*v1.Upstream, error) {
...
+		}
+		if ups.Checks.Active.Unhealthy.Interval == 0 {
+			ups.Checks.Active.Healthy.Interval = 1
```
Comment (gxthrj) [inline pkg/apisix/resource.go:145]: ditto

File: pkg/apisix/upstream.go [hunk 4]
```diff
@@ -62,20 +62,16 @@ func (n *upstreamNodes) UnmarshalJSON(p []byte) error {
...
-	Nodes  upstreamNodes `json:"nodes"`
-	Desc   string        `json:"desc"`
-	Scheme string        `json:"scheme,omitempty"`
+	LBType string                  `json:"type"`
+	HashOn string                  `json:"hash_on,omitempty"`
+	Key    string                  `json:"key,omitempty"`
+	Nodes  upstreamNodes           `json:"nodes"`
+	Desc   string                  `json:"desc"`
+	Scheme string                  `json:"scheme,omitempty"`
+	Checks *v1.UpstreamHealthCheck `json:"checks,omitempty" yaml:"checks,omitempty"`
```
Comment (gxthrj) [inline pkg/apisix/upstream.go:71]: Why declare yaml mark ?

File: pkg/kube/apisix_upstream.go [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+		if config.Unhealthy.HTTPCodes != nil && len(config.Unhealthy.HTTPCodes) < 1 {
+			return nil, &translateError{
+				field:  "healthCheck.passive.unhealthy.httpCodes",
+				reason: "empty",
+			}
+		}
+		passive.Unhealthy.HTTPStatuses = config.Unhealthy.HTTPCodes
+	}
+	return &passive, nil
+}
```
Comment (gxthrj) [inline pkg/kube/apisix_upstream.go:260]: We can use `jsonschema` to simplify validation and avoid too many if else. 
This suggest is an enhancement, do not block the PR.

File: pkg/apisix/resource.go (hunk 2 above)
Comment (tokers) [reply pkg/apisix/resource.go:142]: This is a work around scheme to avoid the APISIX's bug.

File: pkg/kube/apisix_upstream.go (hunk 5 above)
Comment (tokers) [reply pkg/kube/apisix_upstream.go:260]: We can enhance these logics in the future.

File: pkg/apisix/resource.go (hunk 2 above)
Comment (tokers) [reply pkg/apisix/resource.go:142]: The default interval in APISIX is zero...

File: pkg/apisix/upstream.go (hunk 4 above)
Comment (tokers) [reply pkg/apisix/upstream.go:71]: removed.

File: pkg/apisix/resource.go (hunk 3 above)
Comment (tokers) [reply pkg/apisix/resource.go]: changed.

File: pkg/apisix/resource.go (hunk 2 above)
Comment (tokers) [reply pkg/apisix/resource.go]: changed.

File: pkg/apisix/resource.go (hunk 1 above)
Comment (tokers) [reply pkg/apisix/resource.go]: removed.

**Review by gxthrj — APPROVED** [review]

**Pull request merged**